worlds/zork_grand_inquisitor/client.py

```diff
@@ -0,0 +1,193 @@
+    def _cmd_zork(self) -> None:
+        """Attach to an open Zork Grand Inquisitor process."""
+        result: bool = self.ctx.game_controller.open_process_handle()
+
+        if result:
+            self.ctx.process_attached_at_least_once = True
+            self.output("Successfully attached to Zork Grand Inquisitor process.")
+        else:
+            self.output("Failed to attach to Zork Grand Inquisitor process.")
+
+    def _cmd_brog(self) -> None:
+        """List received Brog items."""
+        self.ctx.game_controller.list_received_brog_items()
+
+    def _cmd_griff(self) -> None:
+        """List received Griff items."""
+        self.ctx.game_controller.list_received_griff_items()
+
+    def _cmd_lucy(self) -> None:
+        """List received Lucy items."""
+        self.ctx.game_controller.list_received_lucy_items()
+
+    def _cmd_hotspots(self) -> None:
+        """List received Hotspots."""
+        self.ctx.game_controller.list_received_hotspots()
+
+
+class ZorkGrandInquisitorContext(CommonClient.CommonContext):
+    tags: Set[str] = {"AP"}
+    game: str = "Zork Grand Inquisitor"
+    command_processor: CommonClient.ClientCommandProcessor = ZorkGrandInquisitorCommandProcessor
+    items_handling: int = 0b111
+    want_slot_data: bool = True
+
+    item_name_to_id: Dict[str, int] = item_names_to_id()
+    location_name_to_id: Dict[str, int] = location_names_to_id()
+
+    id_to_items: Dict[int, ZorkGrandInquisitorItems] = id_to_items()
+    id_to_locations: Dict[int, ZorkGrandInquisitorLocations] = id_to_locations()
+
+    game_controller: GameController
+
+    controller_task: Optional[asyncio.Task]
+
+    process_attached_at_least_once: bool
+    can_display_process_message: bool
+
+    def __init__(self, server_address: Optional[str], password: Optional[str]) -> None:
+        super().__init__(server_address, password)
+
+        self.game_controller = GameController(logger=CommonClient.logger)
+
+        self.controller_task = None
+
+        self.process_attached_at_least_once = False
+        self.can_display_process_message = True
+
+    def run_gui(self) -> None:
+        from kvui import GameManager
+
+        class TextManager(GameManager):
+            logging_pairs: List[Tuple[str, str]] = [("Client", "Archipelago")]
+            base_title: str = "Archipelago Zork Grand Inquisitor Client"
+
+        self.ui = TextManager(self)
+        self.ui_task = asyncio.create_task(self.ui.async_run(), name="UI")
+
+    async def server_auth(self, password_requested: bool = False):
+        if password_requested and not self.password:
+            await super().server_auth(password_requested)
+
+        await self.get_username()
+        await self.send_connect()
+
+    def on_package(self, cmd: str, _args: Any) -> None:
+        if cmd == "Connected":
+            self.game = self.slot_info[self.slot].game
+
+        Utils.async_start(process_package(self, cmd, _args))
```

Comment:
it's odd, but certainly not  an error, to have an if  cmd == connected, then  start an async task always for every packet, but that handler only reacts to cmd == Connected too.

Reply:
since the function is doing nothing that can be awaited, I'd just do it right here.